server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -0,0 +1,312 @@
+            .put(NodeDuressSettings.SETTING_HEAP_THRESHOLD.getKey(), 0.0)
+            .put(NodeDuressSettings.SETTING_NUM_SUCCESSIVE_BREACHES.getKey(), 1)
+            .put(SearchShardTaskSettings.SETTING_TOTAL_HEAP_PERCENT_THRESHOLD.getKey(), 0.0)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+    }
+
+    @After
+    public final void cleanupNodeSettings() {
+        assertAcked(
+            client().admin()
+                .cluster()
+                .prepareUpdateSettings()
+                .setPersistentSettings(Settings.builder().putNull("*"))
+                .setTransientSettings(Settings.builder().putNull("*"))
+        );
+    }
+
+    public void testSearchShardTaskCancellationWithHighElapsedTime() throws InterruptedException {
+        Settings request = Settings.builder()
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(ElapsedTimeTracker.SETTING_ELAPSED_TIME_MILLIS_THRESHOLD.getKey(), 1000)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+        ExceptionCatchingListener listener = new ExceptionCatchingListener(latch);
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_ELAPSED_TIME), listener);
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+
+        Exception caughtException = listener.getException();
+        assertNotNull("SearchShardTask should have been cancelled with TaskCancelledException", caughtException);
+        assertEquals(TaskCancelledException.class, caughtException.getClass());
+        assertTrue(caughtException.getMessage().contains("elapsed time exceeded"));
+    }
+
+    public void testSearchShardTaskCancellationWithHighCpu() throws InterruptedException {
+        Settings request = Settings.builder()
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(CpuUsageTracker.SETTING_CPU_TIME_MILLIS_THRESHOLD.getKey(), 1000)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+        ExceptionCatchingListener listener = new ExceptionCatchingListener(latch);
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_CPU), listener);
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
+
+        Exception caughtException = listener.getException();
+        assertNotNull("SearchShardTask should have been cancelled with TaskCancelledException", caughtException);
+        assertEquals(TaskCancelledException.class, caughtException.getClass());
+        assertTrue(caughtException.getMessage().contains("cpu usage exceeded"));
```

Comment:
Another nitpick but you'll get much better error messages with following style:

```
MatcherAssert.assertThat(caughtException, instanceOf(TaskCancalledException.class));
MatcherAssert.assertThat(caughtException.getMessage(), containsString("cpu usage exceeded"));
```

because the error message will show what was found versus what was expected instead of just indicating a failure. Not a big deal though.

Reply:
Got it. Changed it.